src/argilla_llama_index/llama_index_handler.py

```diff
@@ -1,58 +1,31 @@
-from argilla._constants import DEFAULT_API_KEY, DEFAULT_API_URL
-import argilla as rg
-from typing import Any, Dict, List, Optional
+from collections import defaultdict
+from datetime import datetime
+import os
 from packaging.version import parse
+from typing import Any, Dict, List, Optional
 import warnings
-import os
-from datetime import datetime
-from collections import defaultdict
-from contextvars import ContextVar
 
+from argilla_llama_index.helpers import _calc_time, _get_time_diff
+
+import argilla as rg
+from argilla._constants import DEFAULT_API_KEY, DEFAULT_API_URL
 from llama_index.core.callbacks.base_handler import BaseCallbackHandler
 from llama_index.core.callbacks.schema import (
-    BASE_TRACE_EVENT,
     CBEventType,
     EventPayload,
     CBEvent,
 )
 
-global_stack_trace = ContextVar("trace", default=[BASE_TRACE_EVENT])
 
 class ArgillaCallbackHandler(BaseCallbackHandler):
-    """Callback handler for Argilla.
-    
-    Args:
-        dataset_name: The name of the dataset to log the events to. If the dataset does not exist,
-            a new one will be created.
-        number_of_retrievals: The number of retrieved documents to log.
-        workspace_name: The name of the workspace to log the events to.
-        api_url: The URL of the Argilla server.
-        api_key: The API key to use to connect to Argilla.
-        event_starts_to_ignore: A list of event types to ignore when they start.
-        event_ends_to_ignore: A list of event types to ignore when they end.
-        handlers: A list of handlers to run when an event starts or ends.
-    
-    Raises:
-        ImportError: If the `argilla` Python package is not installed or the one installed is not compatible
-        ConnectionError: If the connection to Argilla fails
-        FileNotFoundError: If the retrieval and creation of the `FeedbackDataset` fails
-    
-    Example:
-        >>> from argilla_llama_index import ArgillaCallbackHandler
-        >>> from llama_index import VectorStoreIndex, ServiceContext, SimpleDirectoryReader
-        >>> from llama_index.llms import OpenAI
-        >>> from llama_index import set_global_handler
-        >>> set_global_handler("argilla", dataset_name="query_model")
-        >>> llm = OpenAI(model="gpt-3.5-turbo", temperature=0.8)
-        >>> service_context = ServiceContext.from_defaults(llm=llm)
-        >>> docs = SimpleDirectoryReader("data").load_data()
-        >>> index = VectorStoreIndex.from_documents(docs, service_context=service_context)
-        >>> query_engine = index.as_query_engine()
-        >>> response = query_engine.query("What did the author do growing up dude?")
     """
+    Callback handler that logs predictions to Argilla.
 
-    REPO_URL: str = "https://github.com/argilla-io/argilla"
-    ISSUES_URL: str = f"{REPO_URL}/issues"
+    This handler automatically logs the predictions made with LlamaIndex to Argilla,
```

Comment:
can we add an example usage here?

Reply:
Perhaps we can also add some info about config .

Reply:
I've added this example of usage:

Usage:
    ```python
    from llama_index.core import VectorStoreIndex, ServiceContext, SimpleDirectoryReader, set_global_handler
    from llama_index.llms.openai import OpenAI

    set_global_handler("argilla", dataset_name="query_model")

    llm = OpenAI(model="gpt-3.5-turbo", temperature=0.8, openai_api_key=os.getenv("OPENAI_API_KEY"))

    service_context = ServiceContext.from_defaults(llm=llm)
    docs = SimpleDirectoryReader("../../data").load_data()
    index = VectorStoreIndex.from_documents(docs, service_context=service_context)
    query_engine = index.as_query_engine()

    response = query_engine.query("What did the author do growing up?")
    ```
    
    Should I include the imports, or should we strive for minimal number of lines?